### ISSUE TITLE
Cannot update libphonenumber lib.

### ISSUE THREAD

**Issue by OsaAjani** [jackocnr/intl-tel-input#1198]
Hello, i wanted to update libphonenumber lib in order to get the latest change concerning the recent format change of ivory coast phone numbers, though i cannot update the lib following the guide in CONTRIBUTING.

I've been searching for a while, but i'm no expert with JS and Java compiling tools and i struggle to find a solution.

### Steps to reproduce
1.  Follow the guide https://github.com/jackocnr/intl-tel-input/blob/master/.github/CONTRIBUTING.md#updating-to-a-new-version-of-libphonenumber

### Expected behaviour
We should be able to update the libephonenumber lib.

### Actual behaviour
I got a few different errors :
1. In step 1, the guide say to run `mvn -DskipTests` in closure-compiler directory. But we got the error "No goals have been specified for this build.". More generally it seems closure-compiler now use Bazel instead of Maven. Anyway, i found a way to compile closure-compiler by following the "Building from a terminal" part from the README of the closure-compiler.
2. In step 2, when i run the command `ant -f libphonenumber/javascript/build.xml compile-demo` i got this error. 
```Buildfile: /home/ajani/workspace/libphonenumber-tools/libphonenumber/javascript/build.xml

compile-demo:
     [exec] /home/ajani/workspace/libphonenumber-tools/libphonenumber/javascript/../../closure-library/closure/bin/build/closurebuilder.py: Scanning paths...
     [exec] /home/ajani/workspace/libphonenumber-tools/libphonenumber/javascript/../../closure-library/closure/bin/build/closurebuilder.py: 1656 sources scanned.
     [exec] /home/ajani/workspace/libphonenumber-tools/libphonenumber/javascript/../../closure-library/closure/bin/build/closurebuilder.py: Building dependency tree..
     [exec] /home/ajani/workspace/libphonenumber-tools/libphonenumber/javascript/../../closure-library/closure/bin/build/closurebuilder.py: No namespaces found. At least one namespace must be specified with the --namespace or --input flags.

BUILD FAILED
/home/ajani/workspace/libphonenumber-tools/libphonenumber/javascript/build.xml:99: The following error occurred while executing this line:
/home/ajani/workspace/libphonenumber-tools/libphonenumber/javascript/build.xml:23: exec returned: 2

Total time: 1 second
```

**Comment by OsaAjani** [comment]
To follow up, i've managed to update libphonenumber but it was quite a challenge and the CONTRIBUTING file really need to be updated.

As said, Bazel replace maven for compiling `closure-compiler`. Also i actually used a precompiled .jar instead of compiling it myself. To do so i had to update file `libphonenumber/javascript/build.xml` and modify the property `closure-compiler.jar` to point to the precompiled .jar.

Also, to get rid of the previously quoted error `No namespaces found. At least one namespace must be specified with the --namespace or --input flags.`, i had to add a line `goog.provide('i18n.phonenumbers.demo');` at the beginning of `/libphonenumber/javascript/i18n/phonenumbers/demo.js` after step 1 of CONTRIBUTING.

After compilation still failed because some modules was used but not included. To fix that i had to also add a line `goog.require('i18n.phonenumbers.Error');` to `demo.js` and `i18n/phonenumbers/phonenumber.pb.js` to files `i18n/phonenumbers/phonemetadata.pb.js` and `i18n/phonenumbers/phonenumber.pb.js`.

Hope this can help.

**Comment by jonathan42** [comment]
I found an old issue, with a automatic system
https://github.com/jackocnr/intl-tel-input/pull/1162

it could be good that it is validated @OsaAjani @jackocnr  thx

**Comment by OsaAjani** [comment]
Yes, i've seen this PR, but i honestly just dont understand the all compilation/building process of javascript and Grunt to give any opinion on this. So, it's left to @jackocnr :/

**Comment by cmrd-senya** [comment]
I managed to build new `utils.js` following the contribution guidelines from here https://github.com/jackocnr/intl-tel-input/blob/master/.github/CONTRIBUTING.md#updating-to-a-new-version-of-libphonenumber

But I had to make some updates both from the libphonenumber side and intl-tel-input side.

Here are the two respective PRs:
https://github.com/google/libphonenumber/pull/2592
https://github.com/jackocnr/intl-tel-input/pull/1208

As long as you apply both and follow the guideline from `CONTRIBUTING.md` you'll be able to build new `utils.js` with `ant -f libphonenumber/javascript/build.xml compile-demo` without errors. There will be some warnings still but I haven't gone that far to fix the warnings, I just made it compile successfully.

**Comment by cmrd-senya** [comment]
The new version 8.12.20 of libphonenumber  has been released, it contains the fix that is necessary to get `utils.js` built

**Comment by jonathan42** [comment]
thank you, I submitted an update of lib  #1213

**Comment by jackocnr** [comment]
I've just published a new version (v17.0.13) using a newer version of libphonenumber, which should have fixed the issue. Feel free to post here if you're still seeing the issue tho.
